src/component/menu/ApiDetails/ApiDetails.tsx

```diff
@@ -0,0 +1,106 @@
+import useUiConfig from '../../../hooks/api/getters/useUiConfig/useUiConfig';
+import { ReactElement } from 'react';
+import ConditionallyRender from '../../common/ConditionallyRender';
+import { IVersionInfo } from '../../../interfaces/uiConfig';
+
+interface IPartialUiConfig {
+	name: string;
+	version: string;
+	slogan?: string;
+	environment?: string;
+	versionInfo?: IVersionInfo;
+}
+
+interface IProps {
+	uiConfig: IPartialUiConfig;
+}
+
+export function ApiDetailsContainer(): ReactElement {
+	const { uiConfig } = useUiConfig();
+
+	return <ApiDetails uiConfig={uiConfig} />;
+}
+
+export function ApiDetails(props: IProps): ReactElement {
+	return (
+		<section title="API details">
+			<CurrentVersion {...props} />
+			<UpdateNotification {...props} />
+			<br />
+			<small>{props.uiConfig.slogan}</small>
+			<br />
+			<InstanceId {...props} />
+		</section>
+	);
+}
+
+function InstanceId(props: IProps) {
```

Comment:
I think this is fine in this case, but personally I prefer when a component has it's own file. I think it's either a case of (a) a component is large enough to warrant splitting it out into multiple components, which will then be placed in it's own file or (b) the amount of code is to small to make a meaningful abstraction. In this case I think the code is so small that we could just merge it directly into ApiDetails without the component abstractions. What do you think?

Reply:
Good point. I do like the small helper components, but either way works for me. :)